core/README.md

```diff
@@ -0,0 +1,160 @@
+
+```ts
+const Author = Builder<TAuthor>({
+  generator,
+  transformers,
+}).books([Book.random()]);
+```
+
+At the same time, we also need to tell the transformers that the `books` field can be built. We do that by specifying the field in the `buildFields` option:
+
+```ts
+const transformers = {
+  default: Transformer<TAuthor, TAuthor>('default', {
+    buildFields: ['books'],
+  }),
+  graphql: Transformer<TAuthor, TAuthorGraphql>('graphql', {
+    buildFields: ['books'],
+    addFields: () => ({
+      __typename: 'Author',
+    }),
+  }),
+};
+```
+
+### Transformers
+
+A transformer is used to transform the data defined in the generator (the initial model) to another shape.
+
+There are 3 available transformer types:
+
+- `default`: this is the "identity" transformer. You would use the `build()` method to get the generated data according to the initial shape.
+- `graphql`: this is the transformer for GraphQL data shapes. You would use the `buildGraphql()` method to get the generated data transformed into a GraphQL shape.
+- `rest`: this is the transformer for REST data shapes. You would use the `buildRest()` method to get the generated data transformed into a REST shape.
+
+Defining a `graphql` and/or a `rest` transformer is optional and depends on what the specific data requirements are.
+
+For example, for the `Author` we can define a `graphql` transformer like this:
+
+```ts
+const transformers = {
+  default: Transformer<TAuthor, TAuthor>('default', {
+    buildFields: ['books'],
+  }),
+  graphql: Transformer<TAuthor, TAuthorGraphql>('graphql', {
+    buildFields: ['books'],
+    addFields: () => ({
+      __typename: 'Author',
+    }),
+  }),
+};
+```
+
+Then calling `Author.random().buildGraphql<TAuthorGraphql>()` will generate the data with the `__typename` field.
+
+A transformer can be configured with different options:
+
+- `buildFields`: a list of fields that are defined as other data models. This is necessary so that transformers of these nested models are also executed.
+- `removeFields`: a list of fields that should be removed from the model.
+- `addFields`: a function that should return an object with new fields that needs to be added to the model. The object is then merged with the initial data.
+  The function gets an object argument with the initial `fields` object, in case it's needed to construct the new object.
+- `replaceFields`: a function that should return the fully transformed value. This is useful for example when transforming data into different data types, for example an object into an array of objects.
+  The function gets an object argument with the initial `fields` object, in case it's needed to construct the value.
+
+> The `replaceFields` option is mutually exclusive in regards to the `addFields` and `removeFields` options.
+
+### Presets
+
+Presets are pre-configured models (builders) that can be defined for specific use cases.
```

Comment:
This may be outside the scope of the PR but it's worth mentioning. It's important to document how [presets should be exported as functions to ensure the randomness of data](https://github.com/commercetools/merchant-center-frontend/tree/master/packages-shared/test-data#presets-1).

Reply:
Ah good point, I'll try to document this as well.